### DISCUSSION TITLE
How do I specify AVIF to use AV1 and also how to specify the encoding "effort"?

### DISCUSSION THREAD

**Discussion by FluffyDiscord** [libvips/php-vips#209] [Q&A] [answered]
title

**Comment by jcupitt** [comment, marked as answer]
Hi @FluffyDiscord, the docs are here:

https://www.libvips.org/API/current/VipsForeignSave.html#vips-heifsave

So perhaps:

```php
$image->writeToFile("x.avif", ["effort" => 9]);
```

`.avif` filenames are always encoded with av1. You can use the `encoder` param to set which AV1 encoder you prefer.

**Reply by FluffyDiscord** [reply]
thank you
